### PR TITLE
Factor out dynamic class finder/loader to a dedicated file

### PR DESCRIPTION
**Pull request by CyberDem0n**
It could be reused to do the same for MPP modules/classes.
Ref: #2940 and #2950

### CODE REVIEW

File: patroni/dynamic_loader.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+logger = logging.getLogger(__name__)
+
+
+def iter_modules(package: str) -> List[str]:
+    """Get names of modules from *package*, depending on execution environment.
+
+    .. note::
+        If being packaged with PyInstaller, modules aren't discoverable dynamically by scanning source directory because
+        :class:`importlib.machinery.FrozenImporter` doesn't implement :func:`iter_modules`. But it is still possible to
+        find all potential DCS modules by iterating through ``toc``, which contains list of all "frozen" resources.
```
Comment (hughcapet) [inline patroni/dynamic_loader.py:24]:
```suggestion
        find all potential modules by iterating through ``toc``, which contains list of all "frozen" resources.
```

File: patroni/dcs/__init__.py [hunk 2]
```diff
@@ -124,42 +100,14 @@ def iter_dcs_classes(
 
     :yields: a tuple containing the module ``name`` and the imported DCS class object.
```
Comment (barthisrael) [inline patroni/dcs/__init__.py:101]:
```suggestion
    :returns: an iterator of tuples, each containing the module ``name`` and the imported DCS class object.
```

I wonder if that describes the new situation more precisely? I mean, it's not generating values, but returning a generator now.

**Review by barthisrael — APPROVED** [review]
Looks good to me -- just a couple small comments.

**Comment by barthisrael** [comment]
👍

**Comment by hughcapet** [comment]
👍

**Pull request merged**